tka/builder_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tka
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestAuthorityBuilderAddKey(t *testing.T) {
+	pub, priv := testingKey25519(t, 1)
+	key := Key{Kind: Key25519, Public: pub, Votes: 2}
+
+	a, _, err := Create(&Mem{}, State{
+		Keys:               []Key{key},
+		DisablementSecrets: [][]byte{disablementKDF([]byte{1, 2, 3})},
+	}, priv)
+	if err != nil {
+		t.Fatalf("Create() failed: %v", err)
+	}
+
+	pub2, _ := testingKey25519(t, 2)
+	key2 := Key{Kind: Key25519, Public: pub2, Votes: 1}
+
+	b := a.NewUpdater()
+	b.AddKey(key2)
+	updates, err := b.Finalize(func(update *AUM) error {
+		update.sign25519(priv)
+		return nil
+	})
+	if err != nil {
+		t.Fatalf("Finalize() failed: %v", err)
+	}
+
+	// See if the update is valid by applying it to the authority
+	// + checking if the new key is there.
+	if err := a.Inform(updates); err != nil {
+		t.Fatalf("could not apply generated updates: %v", err)
+	}
+	if _, err := a.state.GetKey(key2.ID()); err != nil {
+		t.Errorf("could not read new key: %v", err)
+	}
+}
+
+func TestAuthorityBuilderRemoveKey(t *testing.T) {
+	pub, priv := testingKey25519(t, 1)
+	key := Key{Kind: Key25519, Public: pub, Votes: 2}
+	pub2, _ := testingKey25519(t, 2)
+	key2 := Key{Kind: Key25519, Public: pub2, Votes: 1}
+
+	a, _, err := Create(&Mem{}, State{
+		Keys:               []Key{key, key2},
+		DisablementSecrets: [][]byte{disablementKDF([]byte{1, 2, 3})},
+	}, priv)
+	if err != nil {
+		t.Fatalf("Create() failed: %v", err)
+	}
+
+	b := a.NewUpdater()
+	b.RemoveKey(key2.ID())
+	updates, err := b.Finalize(func(update *AUM) error {
+		update.sign25519(priv)
+		return nil
+	})
+	if err != nil {
+		t.Fatalf("Finalize() failed: %v", err)
+	}
+
+	// See if the update is valid by applying it to the authority
+	// + checking if the key has been removed.
+	if err := a.Inform(updates); err != nil {
+		t.Fatalf("could not apply generated updates: %v", err)
+	}
+	if _, err := a.state.GetKey(key2.ID()); err != ErrNoSuchKey {
+		t.Errorf("GetKey(key2).err = %v, want %v", err, ErrNoSuchKey)
+	}
+}
+
+func TestAuthorityBuilderSetKeyVote(t *testing.T) {
```

Comment:
If you're keeping the "update vote with meta" optimzation, there should be a test case that covers that.